### DISCUSSION TITLE
Inserting code within err != nil blocks

### DISCUSSION THREAD

**Discussion by 4ydx** [sqlc-dev/sqlc#3154] [Q&A]
Would it be possible to insert specific error handling code into the err != nil blocks?  As in, would it be too invasive a change to maintain?  Would it possibly be something that could become a feature request?

**Comment by Nikola-Milovic** [comment]
I'd extend this question and ask whether it would be possible to extend the codegen and the resulting generated queries. It would be really useful to add tracing, custom error handling, logging and so on.
